Zend/zend_max_execution_timer.c

```diff
@@ -16,32 +16,129 @@
 
 #ifdef ZEND_MAX_EXECUTION_TIMERS
 
+# ifdef __APPLE__
+
+#include <dispatch/dispatch.h>
+#  ifdef ZTS
+#include <pthread.h>
+#  else
+#include "zend_execute.h"
+#  endif
+
+#include "zend.h"
+#include "zend_globals.h"
+
+// macOS doesn't support timer_create(), fallback to Grand Central Dispatch
+
+static inline void zend_max_execution_timer_handler(void *arg)
+{
+#ifdef ZTS
+	pthread_t *tid = (pthread_t *) arg;
+	pthread_kill(*tid, ZEND_MAX_EXECUTION_TIMERS_SIGNAL);
+#else
+	zend_timeout_handler();
+#endif
+}
+
+static inline void zend_max_execution_timer_cancel(void *arg)
+{
+	free(arg);
+}
+
+ZEND_API void zend_max_execution_timer_init(void) /* {{{ */
+{
+	pid_t pid = getpid();
+
+	if (EG(pid) == pid) {
+		return;
+	}
+
+	dispatch_queue_global_t queue = dispatch_get_global_queue(QOS_CLASS_UTILITY, 0);
```

Comment:
quick question: did you try with using your own queue or at least, does it make any meaningful difference in your opinion e.g.

```c
dispatch_queue_attr_t attr = dispatch_queue_attr_make_with_qos_class(DISPATCH_QUEUE_SERIAL, QOS_CLASS_UTILITY, 0);
dispatch_queue_t queue = dispatch_queue_create("net.php.zend_max_execution_timer", attr);
```

Reply:
According to the docs, this queue looks adapted to our use case, but I'm not a specialist in Mac specifics.

Reply:
ACK. Was wondering what is best for, at least, ZTS context.